tests/test-pow.c

```diff
@@ -98,44 +124,61 @@ int main()
         PoWFPGAAccel_Context,
 #endif
     };
+#if defined(ENABLE_STAT)
+    double hashRateArr[POW_TOTAL];
+    int pow_total = POW_TOTAL;
+#else
+    int pow_total = 1;
+#endif
+
 
     for (int idx = 0; idx < sizeof(ImplContextArr) / sizeof(ImplContext); idx++) {
-        printf("%s\n",description[idx]);
+        printf("%s\n", description[idx]);
 
         ImplContext *PoW_Context_ptr = &ImplContextArr[idx];
 
         /* test implementation with mwm = 14 */
         initializeImplContext(PoW_Context_ptr);
         void *pow_ctx = getPoWContext(PoW_Context_ptr, (int8_t *) trytes, mwm);
         assert(pow_ctx);
-        doThePoW(PoW_Context_ptr, pow_ctx);
-        int8_t *ret_trytes = getPoWResult(PoW_Context_ptr, pow_ctx);
-        assert(ret_trytes);
-        PoW_Info pow_info = getPoWInfo(PoW_Context_ptr, pow_ctx);
-        freePoWContext(PoW_Context_ptr, pow_ctx);
-        destroyImplContext(PoW_Context_ptr);
 
-        Trytes_t *trytes_t = initTrytes(ret_trytes, TRANSACTION_TRYTES_LENGTH);
-        assert(trytes_t);
-        Trytes_t *hash_trytes = hashTrytes(trytes_t);
-        assert(hash_trytes);
-        Trits_t *ret_trits = trits_from_trytes(hash_trytes);
-        assert(ret_trits);
+        for (int count = 0; count < pow_total; count++) {
+            doThePoW(PoW_Context_ptr, pow_ctx);
+            int8_t *ret_trytes = getPoWResult(PoW_Context_ptr, pow_ctx);
+            assert(ret_trytes);
+#if defined(ENABLE_STAT)
+            PoW_Info pow_info = getPoWInfo(PoW_Context_ptr, pow_ctx);
+#endif
+
+            Trytes_t *trytes_t = initTrytes(ret_trytes, TRANSACTION_TRYTES_LENGTH);
+            assert(trytes_t);
+            Trytes_t *hash_trytes = hashTrytes(trytes_t);
+            assert(hash_trytes);
+            Trits_t *ret_trits = trits_from_trytes(hash_trytes);
+            assert(ret_trits);
 
-        /* Validation */
-        for (int i = HASH_TRITS_LENGTH - 1; i >= HASH_TRITS_LENGTH - mwm; i--) {
-            assert(ret_trits->data[i] == 0);
+            /* Validation */
+            for (int i = HASH_TRITS_LENGTH - 1; i >= HASH_TRITS_LENGTH - mwm; i--) {
+                assert(ret_trits->data[i] == 0);
+            }
+
+            free(ret_trytes);
+            freeTrobject(trytes_t);
+            freeTrobject(hash_trytes);
+            freeTrobject(ret_trits);
+
```

Comment:
Though the `test-pow` only be used to test one implementation, the `freePoWContext` and `destroyImplContext` should be placed here beacause there is a outer loop to iterate different `ImplContext`.

Reply:
Mmm...... I do not see any problem here.